core/dbt/adapters/relation/factory.py

```diff
@@ -0,0 +1,135 @@
+from dataclasses import replace
+from typing import Dict, Optional, Set, Type
+
+from dbt.contracts.graph.nodes import ModelNode
+from dbt.contracts.relation import ComponentName, RelationType
+from dbt.exceptions import DbtRuntimeError
+
+from dbt.adapters.relation import models
+
+
+class RelationFactory:
+    """
+    Unlike other classes that get used by adapters, this class is not intended to be subclassed. Instead,
+    an instance should be taken from it that takes in all the required configuration (or defaults to
+    what is here).
+    """
+
+    # this configuration should never change
+    backup_suffix: str = "__dbt_backup"
+    intermediate_suffix: str = "__dbt_tmp"
```

Comment:
nit: uppercasing these constants would be more standard stylistically 

Reference: https://stackoverflow.com/a/45702782

Reply:
True, I was thinking of them as default settings, but they should literally never change. I'll make the change.